src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -1,29 +1,143 @@
-import { type ChangeEvent, useCallback, useContext } from 'react'
-import { Skeleton, Typography } from '@mui/material'
+import { memo, type ReactElement, type SyntheticEvent, useCallback, useContext, useMemo } from 'react'
+
+import {
+  Autocomplete,
+  Box,
+  IconButton,
+  InputAdornment,
+  Skeleton,
+  Tooltip,
+  Popper,
+  type PopperProps,
+  type AutocompleteValue,
+  type MenuItemProps,
+  MenuItem,
+} from '@mui/material'
 import { SafeTxContext } from '../../SafeTxProvider'
+import RotateLeftIcon from '@mui/icons-material/RotateLeft'
+import NumberField from '@/components/common/NumberField'
+import { isMultisigExecutionInfo, isTransactionListItem } from '@/utils/transaction-guards'
+import { uniqBy } from 'lodash'
+import useTxQueue, { useQueuedTxByNonce } from '@/hooks/useTxQueue'
+import useSafeInfo from '@/hooks/useSafeInfo'
 import css from './styles.module.css'
+import useAddressBook from '@/hooks/useAddressBook'
+import { getLatestTransactions } from '@/utils/tx-list'
+import { getTransactionType } from '@/hooks/useTransactionType'
+
+const CustomPopper = function (props: PopperProps) {
+  return <Popper {...props} sx={{ width: '300px !important' }} placement="bottom-start" />
+}
+
+const NonceFormOption = memo(function NonceFormOption({
+  nonce,
+  menuItemProps,
+}: {
+  nonce: number
+  menuItemProps: MenuItemProps
+}): ReactElement {
+  const addressBook = useAddressBook()
+  const transactions = useQueuedTxByNonce(nonce)
+
+  const label = useMemo(() => {
+    const [{ transaction }] = getLatestTransactions(transactions)
+    return getTransactionType(transaction, addressBook).text
+  }, [addressBook, transactions])
+
+  return (
+    <MenuItem key={nonce} {...menuItemProps}>
+      {nonce} ({label} transaction)
+    </MenuItem>
+  )
+})
 
 const TxNonce = () => {
-  const { nonce, setNonce, safeTx } = useContext(SafeTxContext)
+  const { page } = useTxQueue()
+  const { safe } = useSafeInfo()
+  const safeNonce = safe.nonce || 0
+  const { nonce, setNonce, safeTx, recommendedNonce } = useContext(SafeTxContext)
   const isEditable = !safeTx || safeTx?.signatures.size === 0
+  const readonly = !isEditable
+
+  const queuedTxs = useMemo(() => {
+    if (!page || page.results.length === 0) {
+      return []
+    }
+
+    const txs = page.results.filter(isTransactionListItem).map((item) => item.transaction)
+
+    return uniqBy(txs, (tx) => {
+      return isMultisigExecutionInfo(tx.executionInfo) ? tx.executionInfo.nonce : ''
+    })
+  }, [page])
+
+  const options = useMemo(() => {
+    return queuedTxs
+      .map((tx) => (isMultisigExecutionInfo(tx.executionInfo) ? tx.executionInfo.nonce : undefined))
+      .filter((nonce) => nonce !== undefined) as number[]
+  }, [queuedTxs])
+
+  // TODO: Add error/warning somewhere
+  const validateInput = (value: string) => {
+    if (!Number.isInteger(value)) {
+      return false
+    } else if (Number(value) < safeNonce) {
+      return false
+    }
+  }
```

Comment:
Previously we displayed error messages but with the new input field there is no room for it.